packages/jsii-rosetta/lib/util.ts

```diff
@@ -24,6 +24,12 @@ export function printDiagnostics(diags: readonly RosettaDiagnostic[], stream: No
   }
 }
 
+export function formatList(xs: string[], n = 5) {
+  const tooMany = xs.length - n;
+
+  return tooMany > 0 ? `${xs.join(', ')} (and ${tooMany} more)` : xs.join(', ');
+}
```

Comment:
Given the usage I see in `snippet-dependencies.ts`, it looks like this function's implementation is not correct.

```suggestion
export function formatList(xs: readonly string[], n = 5) {
  const tooMany = xs.length - n;
  // Limit xs to the desired visible item count
  xs = XS.slice(0, n);

  const suffix = tooMany > 0
    ? ` (and ${suffix} more)`
    : '';

  return `${xs.join(', ')}${suffix}`;
}
```